pkg/aiengine/data.go

```diff
@@ -22,10 +22,11 @@ import (
 )
 
 var (
-	ipcMutex sync.RWMutex
+	ipcMutex    sync.RWMutex
+	serverMutex sync.RWMutex
 )
 
-const ipcPath = "/tmp/spice_ipc.sock"
+var ipcPath = os.TempDir() + "/spice_ipc.sock"
```

Comment:
I think we should make this be `~/.spice/data.ipc` instead of tmp dir.

Reply:
(we also put logs, etc. there)

Reply:
(and it would be similar to the geth model)

Reply:
This is not as clean in my opinion: the temp folder is automatically cleaned (usually on RAM) while .spice would stay. I changed it to ~/spice as requested (it is a unix socket anyway)